src/components/Query.tsx

```diff
@@ -58,7 +63,11 @@ class QueryHandler extends Component<QueryHandlerProps, QueryHandlerState> {
     this.executeQuery();
   }
 
-  componentDidUpdate() {
+  componentDidUpdate(prevProps: QueryHandlerProps) {
+    if (this.props.skip && prevProps.skip !== this.props.skip) {
```

Comment:
If `props.skip` is `true` here, we'll never reach `this.unsubscribe` above 🤔

Reply:
Ah gotcha. So we'd hit a case where we have a current query subscription, `skip` updates to `true`, this block gets hit and `this.unsubscribe` never gets called. What if we just call `this.unsubscribe()` before the early return here?

Reply:
Yea, that'd be equivalent but it should always be preferred to just call execute again since it always unsubscribes and resubscribes if necessary, so that might make things easier on terms of it always aligning the subscription with the props

Reply:
Agreed, good point. Let me do some refactoring and re-push.